Library/Homebrew/formula.rb

```diff
@@ -657,6 +663,29 @@ def skip_clean?(path)
     self.class.skip_clean_paths.include? to_check
   end
 
+  # Sometimes we accidentally install files outside prefix. After we fix that,
+  # users will get nasty link conflict error. So we create a whitelist here to
+  # allow overwriteing certain files. e.g.
+  #   link_overwrite "bin/foo", "lib/bar"
+  #   link_overwrite "share/man/man1/baz-*"
+  def link_overwrite?(path)
+    # Don't overwrite files for a fresh installation.
+    return false unless any_old_version_installed?
```

Comment:
`brew reinstall` will not fail. `brew uninstall; brew install` will. The idea is if we find conflict when we install this formula first time, we can pretty sure conflicted files are from outside of Homebrew. And I feel uncomfortable to nuke users' files.